Mathlib/Data/List/Sort.lean

```diff
@@ -356,6 +356,52 @@ theorem sorted_insertionSort : ∀ l, Sorted r (insertionSort r l)
 
 end TotalAndTransitive
 
+theorem orderedInsert_stable {l c : List α} (a : α) (hl : c <+ l) : c <+ orderedInsert r a l := by
```

Comment:
Here's a shorter proof
```suggestion
theorem orderedInsert_stable {l c : List α} (a : α) (hl : c <+ l) : c <+ orderedInsert r a l  :=
  hl.trans <| sublist_orderedInsert a l
```
which maybe suggests you don't need this lemma at all!